### PR TITLE
Add makeReservation and deleteReservation endpoints

### PR DESCRIPTION
**Pull request by CJ4n**

### CODE REVIEW

File: src/main/java/com/team3/central/controllers/ReservationApiImpl.java [hunk 1]
```diff
@@ -1,9 +1,67 @@
...
+      @RequestHeader(value = "eventId", required = true) Long eventId,
+      @RequestHeader(value = "placeID", required = false) Long placeID) {
+
+    try {
+      Reservation reservation = reservationService.makeReservation(eventId, placeID);
+      ReservationDTO reservationDTO = new ReservationDTO();
+      reservationDTO.setReservationToken(reservation.getReservationToken());
+      reservationDTO.setPlaceId(reservation.getPlaceOnSchema());
+      reservationDTO.setEventId(reservation.getEvent().getId());
+      //TODO: add mapper
```
Comment (adamdab) [inline src/main/java/com/team3/central/controllers/ReservationApiImpl.java]: ?

**Review by adamdab — COMMENTED** [review]
If you explain this to do, it can be merged

**Comment by CJ4n** [comment]
don't merge it, it's not finished yet

**Comment by CJ4n** [comment]
There are some changes to event related classes, but those are necessary for reservations to work properly

File: src/main/java/com/team3/central/services/ReservationService.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      throw new NotFoundException("No such reservation");
+    }
+    reservation.setReservationToken(null);
+    reservationRepository.save(reservation);
+  }
+
+  public Reservation makeReservation(Long eventId, Long placeId)
+      throws NotFoundException, NoFreePlaceException {
+
+    Reservation reservation = reservationRepository.findAll().stream()
```
Comment (adamdab) [inline src/main/java/com/team3/central/services/ReservationService.java]: Can't we use something like findByEventId ?

**Review by adamdab — APPROVED** [review]

**Pull request merged**